### PR TITLE
Move distributed from utils to training

### PR DESCRIPTION
**Pull request by RdoubleA**
#### Context
What is the purpose of this PR? Is it to
- [ ] add a new feature
- [ ] fix a bug
- [ ] update tests and/or documentation
- [ ] other (please add here)

Addresses #1442.

#### Changelog
What are the changes made in this PR?

#### Test plan
Please make sure to do each of the following if applicable to your PR. (If you're not sure about any one of these just ask and we will happily help. We also have a [contributing page](https://github.com/pytorch/torchtune/blob/main/CONTRIBUTING.md) for some guidance on contributing.)

- [ ] run pre-commit hooks and linters (make sure you've first installed via `pre-commit install`)
- [ ] add [unit tests](https://github.com/pytorch/torchtune/tree/main/tests/torchtune) for any new functionality
- [ ] update [docstrings](https://github.com/pytorch/torchtune/tree/main/docs/source) for any new or updated methods or classes
- [ ] run unit tests via `pytest tests`
- [ ] run recipe tests via `pytest tests -m integration_test`
- [ ] manually run any new or modified recipes with sufficient proof of correctness
- [ ] include relevant commands and any other artifacts in this summary (pastes of loss curves, eval results, etc.)

#### UX
If your function changed a public API, please add a dummy example of what the user experience will look like when calling it.
Example of docstring: https://github.com/pytorch/torchtune/blob/6a7951f1cdd0b56a9746ef5935106989415f50e3/torchtune/modules/vision_transformer.py#L285
Example in our docs: https://pytorch.org/torchtune/main/tutorials/qat_finetune.html#applying-qat-to-llama3-models

- [ ] I did not change any public API;
- [ ] I have added an example to docs or docstrings;

### CODE REVIEW

**Review by ebsmothers — APPROVED** [review]
Once CI is green this looks good

File: torchtune/training/_distributed.py [hunk 1]
```diff
@@ -280,7 +280,7 @@ def lora_wrap_fsdp(module: nn.Module, recurse: bool, **kwargs):
 
 
 def load_from_full_model_state_dict(
-    model: "FSDPModule",
+    model: "FSDPModule",  # noqa
```
Comment (felipemello1) [inline torchtune/training/_distributed.py:283]: this seems a bit off, since we only import FSDPModule a few lines later. So it means that it may be something that is not an FSDPModule. Is that right?

Not related to this PR, but maybe the qa shouldnt be supressed.

**Review by felipemello1 — APPROVED** [review]
tests are running still. Approving to unblock. Left just one comment.

File: torchtune/utils/__init__.py [hunk 2]
```diff
@@ -5,23 +5,6 @@
 # LICENSE file in the root directory of this source tree.
 
 from ._device import get_device
-from ._distributed import (  # noqa
-    contains_fsdp,
-    FSDPPolicyType,
```
Comment (felipemello1) [inline torchtune/utils/__init__.py:10]: some of the removed imports are still in __all__

**Review by felipemello1 — CHANGES_REQUESTED** [review]

File: torchtune/utils/__init__.py (hunk 2 above)
Comment (RdoubleA) [reply torchtune/utils/__init__.py:10]: ugh, too many merge conflicts

**Review by felipemello1 — APPROVED** [review]

**Comment by felipemello1** [comment]
approving to unblock

File: torchtune/training/_distributed.py (hunk 1 above)
Comment (RdoubleA) [reply torchtune/training/_distributed.py:283]: Probably because the import line is not at the top so the linter doesn't like this. I'll leave the noqa for now and we can address in a follow up
